### PR TITLE
Feature/get integrated address

### PR DESCRIPTION
**Pull request by BadVoice**
**This feature is based on the zano repository. You can check the repository here:** [https://github.com/hyle-team/zano/blob/fcb5513d6804b15c93a50c3281951cd44b606f5c/src/currency_core/currency_format_utils.cpp#L4249]

I also don't understand why in the zano sources it is checked that flag == 0
` if (addr.flags == 0)
      return tools::base58::encode_addr(CURRENCY_PUBLIC_INTEG_ADDRESS_BASE58_PREFIX, t_serializable_object_to_blob(addr.to_old()) + payment_id); // classic integrated Zano address`

**Important points to note:**
1. Flags - The flags for the integrated address are different from the original zano sources, but it works.
2. Tags - The tags for the integrated address differ from the zano sources as well, but it functions correctly.

You can also look at the tests to understand how it works.

### CODE REVIEW

File: tests/address-utils.spec.ts [hunk 1]
```diff
@@ -12,7 +12,8 @@ describe(
     const address = zanoAddressUtils.encodeAddress(tag, flag, spendPublicKey, viewPublicKey);
 
     it('checking the correctness of the result', () => {
-      expect(address).toBe('ZxD5aoLDPTdcaRx4uCpyW4XiLfEXejepAVz8cSY2fwHNEiJNu6NmpBBDLGTJzCsUvn3acCVDVDPMV8yQXdPooAp338Se7AxeH');
+      expect(address)
```
Comment (Fanaconz) [inline tests/address-utils.spec.ts:15]: why?

File: tests/address-utils.spec.ts [hunk 2]
```diff
@@ -84,30 +86,40 @@ describe(
...
     });
-  });
+  },
+);
 
-describe('Integrated Address Handling', () => {
-  const zanoAddressUtils = new ZanoAddressUtils();
+describe('getIntegratedAddress', () => {
+  const zanoAddressUtils: ZanoAddressUtils = new ZanoAddressUtils();
+  const SUFFIX_LENGTH: number = 18; // paymentId + checksum
```
Comment (Fanaconz) [inline tests/address-utils.spec.ts:94]: why u sum it?

**Review by Fanaconz — APPROVED** [review]
Nice work!

File: tests/address-utils.spec.ts (hunk 1 above)
Comment (BadVoice) [reply tests/address-utils.spec.ts:15]: what why /

File: tests/address-utils.spec.ts (hunk 2 above)
Comment (BadVoice) [reply tests/address-utils.spec.ts:94]: cause it changes with each integrated addresses, we always have a different payment_id and checksum

Comment (BadVoice) [reply tests/address-utils.spec.ts:94]: for convenience, I combined these two parameters into one constant in order to cut them from the final result of the integrated address for tests

Comment (Fanaconz) [reply tests/address-utils.spec.ts:94]: thx, i not have questions )

**Pull request merged**